mmv1/third_party/terraform/website/docs/r/composer_environment.html.markdown

```diff
@@ -993,6 +973,346 @@ The `workloads_config` block supports:
+  and cannot be overridden.
+
+* `pypi_packages` -
+  (Optional)
+  Custom Python Package Index (PyPI) packages to be installed
+  in the environment. Keys refer to the lowercase package name (e.g. "numpy"). Values are the lowercase extras and
+  version specifier (e.g. "==1.12.0", "[devel,gcp_api]", "[devel]>=1.8.2, <1.9.2"). To specify a package without
+  pinning it to a version specifier, use the empty string as the value.
+
+* `env_variables` -
+  (Optional)
+  Additional environment variables to provide to the Apache Airflow scheduler, worker, and webserver processes.
+  Environment variable names must match the regular expression `[a-zA-Z_][a-zA-Z0-9_]*`.
+  They cannot specify Apache Airflow software configuration overrides (they cannot match the regular expression
+  `AIRFLOW__[A-Z0-9_]+__[A-Z0-9_]+`), and they cannot match any of the following reserved names:
+  ```
+  AIRFLOW_HOME
+  C_FORCE_ROOT
+  CONTAINER_NAME
+  DAGS_FOLDER
+  GCP_PROJECT
+  GCS_BUCKET
+  GKE_CLUSTER_NAME
+  SQL_DATABASE
+  SQL_INSTANCE
+  SQL_PASSWORD
+  SQL_PROJECT
+  SQL_REGION
+  SQL_USER
+  ```
+
+* `image_version` -
+  (Required in Cloud Composer 2 and 3, optional in Cloud Composer 1)
+
+  In Cloud Composer 3, you can only specify 3 in Cloud Composer portion of the image version.
+
+  The Apache Airflow portion of the image version is a full semantic version that points to one of the
+  supported Apache Airflow versions, or an alias in the form of only major, major.minor or major.minor.patch versions specified.
+  Like in Composer 1 and 2, a given Airflow version is released multiple times in Composer, with different patches
+  and versions of dependencies. To distinguish between these versions in Composer 3, you can optionally specify a
+  build number to pin to a specific Airflow release.
+  Example: composer-3-airflow-2.6.3-build.4.
+
+  The image version in Composer 3 must match the regular expression:
+  `composer-(([0-9]+)(\.[0-9]+\.[0-9]+(-preview\.[0-9]+)?)?|latest)-airflow-(([0-9]+)((\.[0-9]+)(\.[0-9]+)?)?(-build\.[0-9]+)?)`
+  Example: composer-3-airflow-2.6.3-build.4
+
+  **Important**: In-place upgrade for Composer 3 is not yet supported.
+
+* `cloud_data_lineage_integration` -
+  (Optional, [Beta](https://terraform.io/docs/providers/google/guides/provider_versions.html),
+  Cloud Composer environments in versions composer-2.1.2-airflow-*.*.* and later)
+  The configuration for Cloud Data Lineage integration. Structure is
+  [documented below](#nested_cloud_data_lineage_integration).
+
+* `web_server_plugins_mode` -
+  (Optional, [Beta](https://terraform.io/docs/providers/google/guides/provider_versions.html), Cloud Composer 3 only)
+  Web server plugins configuration. Can be either 'ENABLED' or 'DISABLED'. Defaults to 'ENABLED'.
+
+<a name="nested_cloud_data_lineage_integration"></a>The `cloud_data_lineage_integration` block supports:
```

Comment:
All the blocks like this will need to be disambiguated to specify whether they are composer 1, composer 2, or composer 3 - otherwise links will always take you to the composer 1 version.

Reply:
done.